src/test/java/bio/terra/service/filedata/azure/util/BlobIOTestUtility.java

```diff
@@ -25,19 +27,19 @@
  * copy operations.
  */
 public class BlobIOTestUtility {
+  public static final long MiB = 1024 * 1024;
   private static final String SOURCE_BLOB_NAME = "myTestBlob";
-  static final long MiB = 1024 * 1024;
   private final BlobContainerClient sourceBlobContainerClient;
 
   public BlobContainerClient getSourceBlobContainerClient() {
     return sourceBlobContainerClient;
   }
 
   public BlobContainerClient getDestinationBlobContainerClient() {
-    return destinationBlobContainerClient;
+    return destinationBlobContainerClient.orElseThrow();
```

Comment:
Is the idea that in some cases it will be `null` and that's OK? Otherwise it would be easier to do this test in the constructor.

Reply:
I wanted to make it ok to have a null so the constructor doesn't make sense. The constructor allows a null input and converts it to an optional.  It only fails if you call that method when you've passed in a null

Reply:
So allowing `null` is a new feature you added, I understand. If the point is to fail with an exception at runtime, allowing `null` would work fine here, as there's no benefit to explicitly throwing vs triggering an NPE in test code. I would remove `Optional` and use `null` directly, as it's less code and has the same effect.

Reply:
I feel like having an optional makes it much clearer semantically that it's intended that that field can be validly null...I still contend that all NPEs are uncaught bugs

Reply:
In this code path it's not valid for it to be null. In code paths where it is allowed to be null, it's not used. There's no difference in code behavior or effect, just more code for us to maintain.

Not sure what you mean by NPEs being uncaught bugs. If it's null then the test fails, which we want it to do. I don't think it makes a difference if fails due to an NPE or due to `orElseThrow()`.